src/tokens.c

```diff
@@ -75,9 +77,21 @@ int Token_RecognizeKeyword(char *word, int length)
     char c;
     while(i < length) { /* Put the word through the transition table / DFA*/
         c = word[i];
-        if (c < 'a' || c > 'z') {
-            i = length;
+        if (c >= 'A' && c <= 'Z') {
+            /* if it's a capital letter, make it lowercase. */
+            c = c | 0x60;
+        }
+        if(c >='0' && c <='9') {
+            /* if a number is involved, it's an identifier, because no keywords have numbers. We can already guarantee that it won't start with a number because if it did it would have been already extracted as an int. */
```

Comment:
This will need to change: we should manually change the state in this if block to state 38, the identifier state, so as it loops it can keep checking for syntactically invalid characters, ensuring it winds up in the error state if there is at least one. Right now, a word like `hello7&` would read as a valid identifier when it should result in a token of ERROR.